test/Oscoin/Test/P2P/Handshake.hs

```diff
@@ -133,13 +186,29 @@ tripping' x f g = do
     gf <- g fx
     gf === x
 
-keyPairs :: IsCrypto c => PropertyT IO (Crypto.KeyPair c, Crypto.KeyPair c)
-keyPairs = liftA2 (,) gen gen
-  where
-    gen = evalIO Crypto.generateKeyPair
-
 nidHash
     :: Crypto.Hashable c (PublicKey c)
     => NodeId c
     -> Crypto.Hashed c (PublicKey c)
 nidHash = Crypto.hash . fromNodeId
+
+genSeed :: Gen SplitMix.SMGen
+genSeed = SplitMix.mkSMGen <$> Gen.word64 Range.linearBounded
+
+genKeyPair
+    :: forall c. (HasHashing c, HasDigitalSignature c)
+    => Gen (PublicKey c, PrivateKey c)
+genKeyPair = do
+    seed <- genSeed
+    pure $ withFastRandomBytes seed (generateKeyPair @c)
```

Comment:
For QC, yes. 

The issue is that if you want more than one keypair, the generator needs to use the same seed for all of them. I'm still trying to figure out how to do this best. Someone else is going to be bitten by this again, for sure.

Reply:
Oh, good catch -- this is exactly the same pitfall I also fell into while working on that integration test to generate random ports -- me using `choose (0, x)` meant that the ports were the same for all the 100+ tests as they were all originated by the same seed.

Incidentally, I think I also fell into the same exact pitfall in my patch on the address format, drat!

> I'm still trying to figure out how to do this best.

That is indeed tricky -- if you come across some clever technique keep me posted, as it would be nice to standardise and distill in a separate module all can benefit from.

Reply:
So the best I could come up with for now was [this](https://github.com/oscoin/oscoin/pull/448/files#diff-a15a0d4342481cb0550dbfd07662d05bR71). It's a bit annoying having to pattern match on the result (because `-Wall`).

Reply:
Oh, I see -- essentially, generating a bunch of keys beforehand, and using those in the tests. Yes, I cannot think about any better way either. How annoying!